cpp/include/cugraph/mtmg/sample.cpp

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <raft/core/handle.hpp>
+
+namespace cugraph {
+namespace mtmg {
+
+class handle_t {
+ public:
+  /**
+   * @brief Get the raft handle
+   *
+   * @return const reference to a raft handle
+   */
+  raft::handle_t const& raft_handle() const { return *raft_handle_; }
+
+  /**
+   * @brief Get thread rank
+   *
+   * @return thread rank
+   */
+  int get_thread_rank() const { return thread_rank_; }
```

Comment:
What is the concept of rank ? thread / GPU

Reply:
I think I described this above, so you can read that for a detailed explanation.  The idea is that each CPU thread associated with a particular GPU is numbered (from `[0..m)`).  Mostly we do that to allow for identifying rank 0 as a special thread.  There are (in MPI world, where this concept was borrowed from) some things that you can do with the rank beyond this (you might use the thread rank to break ties, for example).  But I haven't seen anything so far that we would need beyond thread rank 0 and thread rank > 0.